core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/XmlProtocolUnmarshaller.java

```diff
@@ -58,8 +61,16 @@ public static XmlProtocolUnmarshaller create() {
 
     public <TypeT extends SdkPojo> TypeT unmarshall(SdkPojo sdkPojo,
                                                     SdkHttpFullResponse response) {
-        XmlElement document = XmlResponseParserUtils.parse(sdkPojo, response);
-        return unmarshall(sdkPojo, document, response);
+
+        if (hasPayloadMembersOnUnmarshall(sdkPojo)
+            && !hasExplicitBlobPayloadMember(sdkPojo)
+            && !hasExplicitStringPayloadMember(sdkPojo)
+            && response.content().isPresent()) {
+            XmlElement document = XmlResponseParserUtils.parse(sdkPojo, response);
+            return unmarshall(sdkPojo, document, response);
+        } else {
+            return unmarshall(sdkPojo, null, response);
```

Comment:
Because the difference between these two "if" statements is just the value of `document`, can we make that clearer by pulling out the call to `unmarshall`?

```java
XmlElement document = hasXmlPayload(sdkPojo, response) ? XmlResponseParserUtils.parse(sdkPojo, response) : null;
return unmarshall(sdkPojo, document, response);
```

Reply:
+1